File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    s = []
    for i in scoville:
        heapq.heappush(s, i)

    answer = 0
    while s[0] < K or len(s) < 1:
        if len(s) < 2 and s[0] < K:
            return -1
        first = heapq.heappop(s)
        second = heapq.heappop(s)

        new = first + (second * 2)
        heapq.heappush(s, new)
        answer += 1

    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int i = 0; i < scoville.length; i++){
            pq.add(scoville[i]);
        }
        
        while(pq.peek() < K){
            if (pq.peek() < K && pq.size() < 2)
                return -1;
            int min1 = pq.poll();
            int min2 = pq.poll();
            
            int k = min1 + (min2 * 2);
            answer++;
            pq.add(k);
        }
        
        return answer;
    }
}